### PR TITLE
[fix] use expected types

### PR DESCRIPTION
**Pull request by line-o**
- `xs:int` -> `xs:integer`
- cast `xs:boolean` where necessary
- cast `xs:string` to `xs:anyURI` where necessary

### CODE REVIEW

File: src/main/xar-resources/modules/app.xql [hunk 1]
```diff
@@ -108,7 +108,7 @@ function app:module($node as node(), $model as map(*), $details as xs:boolean) {
     
     order by $order
     return
-        app:print-module($module, $funcsInModule, $details)
+        app:print-module($module, $funcsInModule, boolean($details))
```
Comment (joewiz) [inline src/main/xar-resources/modules/app.xql:111]: Is this change necessary? The function signature already defines `$details as xs:boolean`.

File: src/main/xar-resources/modules/scan.xql [hunk 2]
```diff
@@ -40,7 +40,7 @@ declare %private function docs:load-stored($path as xs:anyURI, $store as functio
 declare %private function docs:load-external-modules($store as function(xs:string, element()) as empty-sequence()) {
     for $uri in util:mapped-modules()
```
Comment (joewiz) [inline src/main/xar-resources/modules/scan.xql:41]: (Side note: It would be nice if util:mapped-modules returned xs:anyURI instead of xs:string.)

**Comment by joewiz** [comment]
@line-o Thank you! Could you confirm if you've tested the PR with eXist 7.0.0-SNAPSHOT (develop HEAD)?

File: src/main/xar-resources/modules/app.xql (hunk 1 above)
Comment (line-o) [reply src/main/xar-resources/modules/app.xql:111]: I had to make that change, I believe. Have to check the code.

Comment (joewiz) [reply src/main/xar-resources/modules/app.xql:111]: This change was actually a symptom of a bug in the templating module. I've submitted a PR fixing this: https://github.com/eXist-db/templating/pull/24. I'll merge this now and submit a follow-on PR here once the templating module fix is merged and released.

**Review by joewiz — APPROVED** [review]
